crates/templates/src/source.rs

```diff
@@ -0,0 +1,105 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::{anyhow, Context};
+use tempfile::{tempdir, TempDir};
+use tokio::process::Command;
+use url::Url;
+
+use crate::directory::subdirectories;
+
+const TEMPLATE_SOURCE_DIR: &str = "templates";
+
+#[derive(Debug)]
+pub enum TemplateSource {
+    Git(Url),
+    File(PathBuf),
+}
+
+impl TemplateSource {
+    pub fn try_from_git(git_url: impl AsRef<str>) -> anyhow::Result<Self> {
+        let url_str = git_url.as_ref();
```

Comment:
style: I like shadowing function params when you're just `.as_ref`/`.into`ing them immediately, e.g. `let git_url = git_url.as_ref();`